### ISSUE TITLE
When NHML 1.9.0.11 and 12 benchmarks or mining bminer and gminer it grabs wrong GPU 

### ISSUE THREAD

**Issue by frazier345** [nicehash/NiceHashMiner#1373]
When NHML is benchmarking or mining, it grabs the wrong GPU number So I have 1 1070 and 5 750 Tis on one rig.  It will say it is bench-marking GPU 5 (750 Ti) but the numbers are for the 1070.  When I go to mine it runs the 1070 under Bminer and another miner and does not run GPU 5 at all. Somewhere it is picking the GPU slot number instead of the GPU number in Nice Hash.

**Comment by Oink70** [comment]
Same here... 
GTX1080ti as GPU#1, GTX1050 as GPU#2
On benchmark for GPU#1 the benchmark terminates within seconds.
On benchmark for GPU#2 the benchmark shows the hashrate for GPU#1 (3rd party software shows the benchmark is run on GPU#1)

When mining hashrates drops on GPU#1 (2 miners on 1 GPU) and GPU#2 idling.

NHML v1.9.0.11, clean install on Win10. Bminer and GPU perf monitored with GPU-Z 2.16.0

**Comment by frazier345** [comment]
Also effecting gminer grabs gpu 4 the 1070 instead of GPU 0 the 750 ti.
`[2019-02-04 10:36:46] [INFO] [gminer-MINER_ID(12)-DEVICE_IDs(1,2,3,4,5)] Starting miner [pid(6780)|bin(bin_3rdparty\gminer\miner.exe)] -a 144_5 -s zhash.usa.nicehash.com -n 3369 -u 174xGU8o4axrax417kwMn7hygRR187zo4w.worker1 -d 1 2 3 4 5 --api 4001  --pers auto
[2019-02-04`

**Comment by Oink70** [comment]
Confirmed. Actually, Gminer.exe uses a different way for numbering the GPU's. Since I only have 2 GPU's, there's no way for me to test the logic in their numbering.

In all programs I've used till now, my GPU's have always been ordered like this:

- GPU0: NVIDIA GeForce GTX 1080 Ti 11GB
- GPU1: GIGABYTE GeForce GTX 1050

First lines after starting Gminer manually reveals that Gminer.exe has some reason to number them differently:

```
e:\nhm_windows_1.9.0.11\bin_3rdparty\gminer>miner.exe --algo 150_5 --server beam.eu.nicehash.com --port 3370 --ssl 0 --user 389KMyKSPQEuMJckpNtGHmFceKbcVU9WAM.Oink1
+----------------------------------------------------------------+
|                          GMiner v1.28                          |
+----------------------------------------------------------------+
Algorithm:          Equihash 150,5 "Beam-PoW"
Stratum server:
  host:             stratum://beam.eu.nicehash.com:3370
  user:             389KMyKSPQEuMJckpNtGHmFceKbcVU9WAM.Oink1
  password:         x
Power calculator:   on
Color output:       on
Watchdog:           on
API:                off
Log to file:        off
Selected devices:   GPU0 GPU1
Temperature limits: 90C  90C
------------------------------------------------------------------
16:40:14 Failed to initialize miner on GPU0: GIGABYTE GeForce GTX 1050 2GB: out of memory
16:40:15 Connected to beam.eu.nicehash.com:3370
16:40:15 Authorized on Stratum Server
16:40:15 New Job: 1732693553234 Difficulty: 117440512
16:40:15 Started Mining on GPU1: NVIDIA GeForce GTX 1080 Ti 11GB```

**Comment by Spin55555** [comment]
I confirm the previous users. Will there be a change in NHML to work correctly with Gminer? Or should contact the author of Gminer?

**Comment by Oink70** [comment]
I just noticed NHML 1.9.0.13 is available. Haven't tested on that one yet. I have tested  _NiceHash Miner Legacy Fork Fix 13_ and that one works OK, so contacting author of GMiner seems abundant.

**Comment by tristri** [comment]
nope still not working in 1.9.0.13 gminer still grab the wrong gpu...

**Comment by MindurMiners** [comment]
Still not working.   Anything for me over five cards.   When I try seven only five are actually hashing.....

**Comment by woodyalien** [comment]
same issue here, even on v15

**Comment by S74nk0** [comment]
New version released [1.9.1.5](https://github.com/nicehash/NiceHashMinerLegacy/releases/tag/1.9.1.5) closing this issue. If the issue is present in the new release please open a new issue.

Closing.
